source/ingress.go

```diff
@@ -48,23 +68,49 @@ func (sc *ingressSource) Endpoints() ([]*endpoint.Endpoint, error) {
 	endpoints := []*endpoint.Endpoint{}
 
 	for _, ing := range ingresses.Items {
+		// Check controller annotation to see if we are responsible.
+		controller, exists := ing.Annotations[controllerAnnotationKey]
+		if exists && controller != controllerAnnotationValue { //TODO(ideahitme): log the skip
+			continue
+		}
+
 		ingEndpoints := endpointsFromIngress(&ing)
+
+		// apply template if host is missing on ingress
+		if len(ingEndpoints) == 0 && sc.fqdntemplate != nil {
+			ingEndpoints = sc.endpointsFromTemplate(&ing)
+		}
+
 		endpoints = append(endpoints, ingEndpoints...)
 	}
 
 	return endpoints, nil
 }
 
-// endpointsFromIngress extracts the endpoints from ingress object
-func endpointsFromIngress(ing *v1beta1.Ingress) []*endpoint.Endpoint {
+func (sc *ingressSource) endpointsFromTemplate(ing *v1beta1.Ingress) []*endpoint.Endpoint {
 	var endpoints []*endpoint.Endpoint
 
-	// Check controller annotation to see if we are responsible.
-	controller, exists := ing.Annotations[controllerAnnotationKey]
-	if exists && controller != controllerAnnotationValue {
-		return endpoints
+	var buf bytes.Buffer
+
+	if sc.fqdntemplate.Execute(&buf, ing) == nil { //TODO(ideahitme): if error is present skip or abort ?
```

Comment:
yes, that's why there is TODO, we are lacking logging where it should be logged in multiple places, but I will add the logging here :)